cirq/study/sweepable.py

```diff
@@ -16,17 +16,27 @@
 
 from typing import Dict, Iterable, Iterator, List, Sequence, Union, cast
 import warnings
+from typing_extensions import Protocol
 
 from cirq._doc import document
 from cirq.study.resolver import ParamResolver, ParamResolverOrSimilarType
 from cirq.study.sweeps import ListSweep, Points, Sweep, UnitSweep, Zip, dict_to_product_sweep
 
 SweepLike = Union[ParamResolverOrSimilarType, Sweep]
-document(SweepLike, """An object similar to an iterable of parameter resolvers.""")  # type: ignore
+document(SweepLike, """An object similar to an iterable of parameter resolvers.""")
 
-Sweepable = Union[SweepLike, Iterable[SweepLike]]
+
+class SweepLikeRecursive(Protocol):
+    """An intermediate class allowing for recursive definition of Sweepable,
+    since recursive union definitions are not yet supported in mypy."""
```

Comment:
I agree with not changing the name of `Sweepable`. In that case I think this class should just be named `_Sweepable` to make its purpose clearer.

Reply:
Sounds good, I should have that later today.

Reply:
This is done in latest commit.